File: 8kyu 1/arrayPlusArray.js
/*
|--------------------------------------------------------------------------
| Array plus array
|--------------------------------------------------------------------------
|
*/

function arrayPlusArray(arr1, arr2) {
   return arr1.reduce((acc, curr) => acc + curr) + arr2.reduce((acc, curr) => acc + curr);
}


function arrayPlusArray(arr1, arr2) {
   return arr1.concat(arr2).reduce((acc, cur) => acc + cur);
}


function arrayPlusArray(arr1, arr2) {
   let arr = [...arr1, ...arr2];
   return arr.reduce((a, b) => a + b);
}


function arrayPlusArray(arr1, arr2) {
   let r = 0;
   for (let i = 0; i < arr1.length; i++) {
      r += arr1[i] + arr2[i];
   }
   return r;
}


function arrayPlusArray(...arrays) {
   return [].concat(...arrays).reduce((a, b) => a + b, 0)
}


console.log(arrayPlusArray([1, 2, 3], [4, 5, 6]), 21);
console.log(arrayPlusArray([-1, -2, -3], [-4, -5, -6]), -21);
console.log(arrayPlusArray([0, 0, 0], [4, 5, 6]), 15);
console.log(arrayPlusArray([100, 200, 300], [400, 500, 600]), 2100);

File: Python/8py 1/array_plus_array.py
"""
|--------------------------------------------------------------------------
| Array plus array
|--------------------------------------------------------------------------
|
"""

import functools
def array_plus_array(arr1,arr2):
   return(functools.reduce(lambda a, b: a+b, arr1) + functools.reduce(lambda a, b: a+b, arr2))


def array_plus_array(arr1,arr2):
   return sum(arr1+arr2)


def array_plus_array(arr1,arr2):
   counter = 0
   for i in arr1:
      counter += i
   for i in arr2:
      counter += i
   return counter


def array_plus_array(*args):
   return sum(map(sum, args))


def array_plus_array(arr1,arr2):
   result=0
   for ar1,ar2 in zip(arr1,arr2):
      result = result + (ar1+ar2)
   return result


from itertools import chain
def array_plus_array(*arrays):
   return sum(chain.from_iterable(arrays))


print(array_plus_array([1, 2, 3], [4, 5, 6]), 21)
print(array_plus_array([-1, -2, -3], [-4, -5, -6]), -21)
print(array_plus_array([0, 0, 0], [4, 5, 6]), 15)
print(array_plus_array([100, 200, 300], [400, 500, 600]), 2100)